compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3327,6 +3446,8 @@ static TR::Register *inlineArrayCmp(TR::Node *node, TR::CodeGenerator *cg)
 TR::Register *OMR::Power::TreeEvaluator::arraycmpEvaluator(TR::Node *node, TR::CodeGenerator *cg)
    {
    TR::Compilation *comp = cg->comp();
+   if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P10) && (feGetEnv("TR_DisableP10ArrayCmp") == NULL))
+      return inlineArrayCmpP10(node, cg);
```

Comment:
This is a temporary working version due to ISA limitation we cannot get around.  We will rework it once new ISA replacement comes in.

Reply:
If we aren't sure that this will work on 32-bit, I'd prefer to err on the safe side and gate this new feature behind a check to prevent it from being used on 32-bit until/unless this is confirmed to work there. I doubt many people would try to use a ppc32 version of OMR on P10, but I'd prefer to have this corner case at least be conservatively correct.

Reply:
this will work to the extent we intended.  as long as the resulting registers don't survive across a signal handling boundary, we are fine.

Reply:
Alright, I'm fine with this so long as due diligence has been done on your part to ensure that this functions correctly and this wasn't just left enabled for ppc32 on the assumption the check wasn't needed.